R/orderly_version.R

```diff
@@ -353,6 +366,31 @@ orderly_version <- R6::R6Class(
       session$git <- session$meta$git
       session$archive_version <- cache$current_archive_version
       saveRDS(session, path_orderly_run_rds(private$workdir))
+    },
+
+    write_orderly_fail_rds = function(error) {
+      if (is.null(private$workdir)) {
+        message("Can't save fail RDS, workdir not set")
+        return(invisible(FALSE))
+      }
+      session <- withr::with_envvar(private$envvar, session_info())
+      trace <- utils::limitedLabels(sys.calls())
+      if (length(trace) > 4) {
+        ## Remove the last 4 entries, these will always be calls to:
+        ## .handleSimpleError(...                  # nolint
+        ## h(simpleError(msg, call))               # nolint
+        ## self$run_failed_cleanup(e)              # nolint
+        ## private$write_orderly_fail_rds(error)   # nolint
+        ## which aren't really interesting
+        trace <- trace[seq(1, length(trace) - 4)]
+      }
+      error$trace <- trace
```

Comment:
I know that this is how we do it in rrq, but I wonder about making the trace more discoverable by saving a list of `list(error = error, trace = trace)` or saving trace at the to level too? Otherwise the default print method of error makes you forget there is a trace there

Reply:
Yeah that sounds good